BaseBuilder.cpp

```diff
@@ -83,14 +83,13 @@ void CreateBeacons(int numBeacons, const BeaconInfo beacon[])
 
 void CreateInitialCombatUnits(int playerNum, int x, int y)
 {
-	int numUnits = TethysGame::InitialUnits();		// Number of combat units to start with
-	map_id weaponType = (Player[playerNum].IsEden() == 0) ? mapMicrowave : mapLaser;	// Set weapon based on colony type
+	// Set weapon based on colony type
+	const map_id weaponType = (Player[playerNum].IsEden() == 0) ? mapMicrowave : mapLaser;
 
-	for (; numUnits > 0; --numUnits)
+	Unit unit;
+	for (int i = TethysGame::InitialUnits(); i > 0; --i)
 	{
-		Unit unit;

```

Comment:
Why was `Unit` moved outside the for loop here? This looks like an accidental move.